pkg/kubelet/kubelet.go

```diff
@@ -1224,7 +1226,11 @@ func ensureHostsFile(fileName string, hostIP, hostName string) error {
 	buffer.WriteString("fe00::0\tip6-mcastprefix\n")
 	buffer.WriteString("fe00::1\tip6-allnodes\n")
 	buffer.WriteString("fe00::2\tip6-allrouters\n")
-	buffer.WriteString(fmt.Sprintf("%s\t%s\n", hostIP, hostName))
+	if len(hostDomainName) > 0 {
```

Comment:
here: https://github.com/kubernetes/kubernetes/pull/20688/files#diff-3d761026fe20ec73f4c03d8b0a32d351R132

let me know if you had something else in mind.